src/portfft/committed_descriptor_impl.hpp

```diff
@@ -517,14 +528,34 @@ class committed_descriptor_impl {
 
       auto in_bundle = sycl::get_kernel_bundle<sycl::bundle_state::input>(queue.get_context(), ids);
 
-      set_spec_constants(top_level, in_bundle, length, factors, detail::elementwise_multiply::NOT_APPLIED,
+      if (factor_size != static_cast<Idx>(params.lengths[dimension_num]) && !is_global) {
+        in_bundle.template set_specialization_constant<detail::SpecConstFFTAlgorithm>(detail::fft_algorithm::BLUESTEIN);
```

Comment:
Bluestein cannot be a kernel of its own, so our wi, sg,wg implementation can be seen as the following sequence of steps - 

1. Load data from global to local memory
2. Load data from local to private
3. Perform all calculations required
4. Store data from private to local
5. Store data from local to private

Step number 3 id where bluestein fits in, because that's all it changes, the computation steps. and we can reuse the existing infrastructure we have for steps 1,2,4 and 5

That's also how I was able to so easily support different layouts and complex storage schemes and also why multi-dimensional Bluestein FFTs worked out of the box as well